library/src/animationManager.cxx

```diff
@@ -283,10 +282,83 @@ bool animationManager::LoadAtTime(double timeValue)
   return true;
 }
 
+// ---------------------------------------------------------------------------------
+void animationManager::CycleAnimation()
+{
+  assert(this->Importer);
+  if (this->AvailAnimations <= 0)
+  {
+    return;
+  }
+
+  this->AnimationIndex += 1;
+
+  if (this->AnimationIndex == this->AvailAnimations)
+  {
+    this->AnimationIndex = -1;
+  }
+
+  this->EnableOnlyCurrentAnimation();
+  this->LoadAtTime(this->TimeRange[0]);
+}
+
+// ---------------------------------------------------------------------------------
+int animationManager::GetAnimationIndex()
+{
+  return this->AnimationIndex;
+}
+
+// ---------------------------------------------------------------------------------
+std::string animationManager::GetAnimationName()
+{
+  if (!this->Importer)
+  {
+    return "";
+  }
+
+  if (this->AnimationIndex == -1)
+  {
+    return "All Animations";
+  }
+  return this->Importer->GetAnimationName(this->AnimationIndex);
+}
+
+//----------------------------------------------------------------------------
+void animationManager::EnableOnlyCurrentAnimation()
+{
+  assert(this->Importer);
+  if (this->AnimationIndex - 1 == -1)
+  {
+    this->AnimationIndex = this->AnimationIndex - 1;
+    for (int i = 0; i < this->AvailAnimations; i++)
+    {
+      this->Importer->DisableAnimation(i);
+    }
+    this->AnimationIndex = this->AnimationIndex + 1;
+  }
+  else
+  {
+    this->Importer->DisableAnimation(this->AnimationIndex - 1);
+  }
```

Comment:
this overly complicates things, just start by disabling all animations.

Reply:
Okay fixed. EnableOnlyCurrentAnimation() is essentially:
- DisablePreviousAnimation()
- EnableCurrentAnimation()

?